### PR TITLE
DateTime Support on the Client Side Corresponding to the Server Side DateTime Support

### PR DESCRIPTION
**Pull request by jingjingrong**
### Issues
*This pull request fixes issue #473.*  

### Description
Our team has encountered the same blocking issue when trying to upgrade from OData v3 to v4. Our existing base object already uses DateTime in production, therefore it would be a breaking change during the upgrade if there is no DateTime support. It would be great to find a workaround on the client side without changing the odata V4 protocol.

from OData/WebApi#136 and also http://odata.github.io/WebApi/datetime-support, seems DateTime is officially supported by the OData server side while still maintain the compatibility with the OData v4 protocol, why we could not do the same for client side?

In other words, for the EDM model and over the wire, still use DateTimeOffset to be compatible with OData v4 protocol, while internally convert the DateTimeOffset from/to DateTime, to allow the actual C# object used in client side still be DateTime.

### Checklist (Uncheck if it is not completed)
- [ x ] Test cases added
- [ x ] Build and test with one-click build and test script passed

### Additional work necessary
*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

**Comment by jingjingrong** [comment]
Closed as a duplicated pull request as 715
